test/e2e/targets/googlecloudstorage/main.go

```diff
@@ -0,0 +1,259 @@
+	e2estorage "github.com/triggermesh/triggermesh/test/e2e/framework/gcloud/storage"
+)
+
+/* This test suite requires:
+
+   - A Google Cloud Service Account key in JSON format, exported in the environment as GCLOUD_SERVICEACCOUNT_KEY
+   - The name of the Google Cloud project exported in the environment as GCLOUD_PROJECT
+*/
+
+var targetAPIVersion = schema.GroupVersion{
+	Group:   "targets.triggermesh.io",
+	Version: "v1alpha1",
+}
+
+const (
+	targetKind     = "GoogleCloudStorageTarget"
+	targetResource = "googlecloudstoragetargets"
+)
+
+const gcpCredentialsJosnSecretKey = "credentialsJson"
+
+var _ = Describe("Google Cloud Storage target", func() {
+	// NOTE: bucket names aren't allowed to contain "goog"
+	f := framework.New("gcloudstoragetarget")
+
+	var ns string
+
+	var trgtClient dynamic.ResourceInterface
+
+	var bucketName string
+	var gcpSecret *corev1.Secret
+
+	BeforeEach(func() {
+		ns = f.UniqueName
+
+		gvr := targetAPIVersion.WithResource(targetResource)
+		trgtClient = f.DynamicClient.Resource(gvr).Namespace(ns)
+	})
+
+	Context("a target is deployed", func() {
+		var trgtURL *url.URL
+		var storageClient *storage.Client
+
+		var sentEvent *cloudevents.Event
+
+		var serviceaccountKey string
+		var gcloudProject string
+
+		BeforeEach(func() {
+			var err error
+			serviceaccountKey = e2egcloud.ServiceAccountKeyFromEnv()
+			gcloudProject = e2egcloud.ProjectNameFromEnv()
+
+			storageClient, err = storage.NewClient(context.Background(), option.WithCredentialsJSON([]byte(serviceaccountKey)))
+			Expect(err).ToNot(HaveOccurred())
+
+			gcpSecret = createGCPCredsSecret(f.KubeClient, ns, serviceaccountKey)
+
+			By("creating a Google Cloud Storage Bucket", func() {
+				bucketName = e2estorage.CreateBucket(storageClient, gcloudProject, f)
+
+				DeferCleanup(func() {
+					By("deleting Google Cloud Storage Bucket "+bucketName, func() {
+						e2estorage.DeleteObjects(storageClient, bucketName)
+						e2estorage.DeleteBucket(storageClient, bucketName)
+					})
+				})
+			})
+		})
+
+		When("the spec contains default settings", func() {
+			BeforeEach(func() {
+				By("creating an GoogleCloudStorageTarget object", func() {
+					trgt, err := createTarget(trgtClient, ns, "test-",
+						withBucketName(bucketName),
+						withCredentials(gcpSecret.Name),
+					)
+					Expect(err).ToNot(HaveOccurred())
+
+					trgt = ducktypes.WaitUntilReady(f.DynamicClient, trgt)
+
+					trgtURL = ducktypes.Address(trgt)
+					Expect(trgtURL).ToNot(BeNil())
+				})
+			})
+
+			When("an event is sent to the target", func() {
+
+				BeforeEach(func() {
+					By("sending an event", func() {
+						sentEvent = e2ece.NewHelloEvent(f)
+
+						job := e2ece.RunEventSender(f.KubeClient, ns, trgtURL.String(), sentEvent)
+						apps.WaitForCompletion(f.KubeClient, job)
+					})
+				})
+
+				It("creates an object onto the bucket", func() {
+					var receivedObj []byte
+					var err error
```

Comment:
Not needed outside of the `By`.

Reply:
The recevieObj is used in the other `By`, so I think it's needed to be declared there

Reply:
My comment was on the line that reads
```
var err error
```